channel/consensus_channel/consensus_channel.go

```diff
@@ -624,19 +620,19 @@ func (p SignedProposal) SortInfo() (types.Destination, uint64) {
 }
 
 // Target returns the target channel of the proposal.
-func (p *Proposal) Target() types.Destination {
+func (p *Proposal) Target() (types.Destination, error) {
 	switch p.Type() {
 	case "AddProposal":
 		{
-			return p.ToAdd.Target()
+			return p.ToAdd.Target(), nil
 		}
 	case "RemoveProposal":
 		{
-			return p.ToRemove.Target
+			return p.ToRemove.Target, nil
 		}
 	default:
 		{
-			panic("invalid proposal type")
```

Comment:
This one is potentially debatable. If we hit this line, something has gone very badly wrong and it is definitely a programming error rather than something unexpected in the environment. The data in the store is in an invalid state. So one could argue that panicking is acceptable here.